image-registry/remove-registry-baremetal.md

```diff
@@ -0,0 +1,189 @@
+
+## Summary
+
+On platforms which do not provide shareable object storage, the OpenShift image
+registry operator will bootstrap itself as `Removed`. This will allow
+`openshift-installer` to complete installations on these platform types. A
+separate process will be responsible for configuring storage and updating the
+image registry operator's state to `Managed`.
+
+## Motivation
+
+IPI installation options for Kubernetes-native infrastructure (formerly
+RHHI.next) and Red Hat Virtualization (RHV) have been declared a key initiative
+for OpenShift 4.3. These will configure themselves as the `baremetal` and
+`oVirt` platform types, respectively. In 4.2 the registry bootstraps itself to
+use `EmptyDir` storage on `baremetal` to enable pre-GA development - see
+([PR #332](https://github.com/openshift/cluster-image-registry-operator/pull/332)).
+For RHV no storage is configured and and IPI installation for the `oVirt`
+platform type fails to complete.
+
+IPI installation for `vSphere` is not supported at present, but is desired in
+the future. Current UPI installs on `vSphere` often result in the registry not
+being configured and reporting itself `Degraded=true`.
+
+`EmptyDir` is not suitable for GA components as there is a serious risk of data
+loss if a permanent form of storage is not configured.
+
+### Goals
+
+- `openshift-installer` completes if the platform is set to `baremetal` or
+  `oVirt`.
+- UPI installs for `vSphere` bring the registry up in the `Removed` state.
+- Image registry is able to bring itself up after the management state is
+  switched from `Removed` to `Managed`.
+
+### Non-Goals
+
+- Provision storage for the registry for RHV and k8s-native infrastructre.
+- Provide 1st class support for additional storage types/providers for the
+  registry (ex: MCG).
+- Harden OpenShift features (ex: Builds, ImageStreams) such that they error
+  cleanly if the registry is not installed.
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal actually is.
+
+### User Story
+
+As an OpenShift cluster admin/installer
+I want the registry to be `Removed` on RHV and k8s-native infra installations
+So that I can provision other storage providers
+And later enable the image registry with the desired storage
+
+### Implementation Details/Notes/Constraints
+
+- When the image registry operator bootstraps itself, it will inspect the
+  cluster infrastructure platform status.
+- The registry will bootstrap itself as `Removed` on the following platform types:
+  - `BareMetal`
+  - `oVirt`
+  - `VSphere`
+  - `None`
+- The registry will bootstrap itself as `Managed` with `EmptyDir` storage on
+  the following platform types:
+  - `Libvirt`
+  - `Unknown` platforms
+- The registry will continue to bootstrap itself as `Managed` with specific
+  storage on the following platforms:
+  - `AWS` (Amazon S3)
+  - `Azure` (Azure blob storage)
+  - `GCP` (Google Cloud Storage)
+  - `OpenStack` (Swift)
```

Comment:
@ricardomaraschini note that for now I'm keeping `Libvirt` and `Unknown` to bootstrap as `EmptyDir`. IIRC from our conversations `None` is a true UPI bare-metal installation, whereas `BareMetal` is a managed product like RHHI.next

Reply:
A little bit misguiding the naming. I will adjust the patch accordingly.